.rhdh/scripts/prepare-restricted-environment.sh

```diff
@@ -162,57 +163,107 @@ RELATED_IMAGES=()
-      shift 1;;
-
-    # New options
-    '--index-image') INDEX_IMAGE="$2"; shift 1;;
-    '--ci-index') IS_CI_INDEX_IMAGE="$2"; shift 1;;
-    '--filter-versions')
-      if [[ "$2" == "*" ]]; then
-        NO_VERSION_FILTER="true"
-      else
-        IFS=',' read -r -a FILTERED_VERSIONS <<< "$2"
-      fi
-      shift 1;;
-    '--extra-images') IFS=',' read -r -a EXTRA_IMAGES <<< "$2"; shift 1;;
-    '--to-registry') TO_REGISTRY="$2"; shift 1;;
-    '--to-dir') TO_DIR=$(realpath "$2"); shift 1;;
-    '--from-dir') FROM_DIR="$2"; shift 1;;
-    '--install-operator') INSTALL_OPERATOR="$2"; shift 1;;
-    '--use-oc-mirror') USE_OC_MIRROR="$2"; shift 1;;
-    '--oc-mirror-path') OC_MIRROR_PATH="$2"; shift 1;;
-    '--oc-mirror-flags') OC_MIRROR_FLAGS="$2"; shift 1;;
-    '--install-yq') INSTALL_YQ=1;;
-    '-h'|'--help') usage; exit 0;;
-    *) errorf "Unknown parameter is used: $1."; usage; exit 1;;
+  # Legacy options. Deprecated but kept for backward compatibility
+  '--prod_operator_index')
+    INDEX_IMAGE="$2"
+    shift 1
+    ;;
+  '--prod_operator_package_name')
+    debugf "--prod_operator_package_name ($2) is no longer used"
+    shift 1
+    ;;
+  '--prod_operator_bundle_name')
+    debugf "--prod_operator_bundle_name ($2) is no longer used"
+    shift 1
+    ;;
+  '--helper_mirror_registry_storage')
+    debugf "--helper_mirror_registry_storage is no longer used. This script assumes you already have a mirror registry in your disconnected environment"
+    shift 1
+    ;;
+  '--use_existing_mirror_registry')
+    debugf "--use_existing_mirror_registry is no longer used. This script assumes you already have a mirror registry in your disconnected environment"
+    shift 1
+    ;;
+  '--prod_operator_version')
+    input="${2#v}"
+    IFS='.' read -ra parts <<<"$input"
+    length=${#parts[@]}
+    if [ $length -ge 2 ]; then
+      FILTERED_VERSIONS=("${parts[0]}"."${parts[1]}")
+    else
+      FILTERED_VERSIONS=("${parts[*]}")
+    fi
+    debugf "${FILTERED_VERSIONS[@]}"
+    shift 1
+    ;;
+
+  # New options
+  '--index-image')
+    INDEX_IMAGE="$2"
+    shift 1
+    ;;
+  '--ci-index')
+    IS_CI_INDEX_IMAGE="$2"
+    shift 1
+    ;;
+  '--filter-versions')
+    if [[ "$2" == "*" ]]; then
+      NO_VERSION_FILTER="true"
+    else
+      IFS=',' read -r -a FILTERED_VERSIONS <<<"$2"
+    fi
+    shift 1
+    ;;
+  '--extra-images')
+    IFS=',' read -r -a EXTRA_IMAGES <<<"$2"
+    shift 1
+    ;;
+  '--to-registry')
+    TO_REGISTRY="$2"
+    shift 1
+    ;;
+  '--to-dir')
+    TO_DIR=$(realpath "$2")
+    shift 1
+    ;;
+  '--from-dir')
+    FROM_DIR="$2"
+    shift 1
+    ;;
+  '--install-operator')
+    INSTALL_OPERATOR="$2"
+    shift 1
+    ;;
+  '--use-oc-mirror')
+    USE_OC_MIRROR="$2"
+    shift 1
+    ;;
+  '--oc-mirror-path')
+    OC_MIRROR_PATH="$2"
+    shift 1
+    ;;
+  '--oc-mirror-flags')
+    OC_MIRROR_FLAGS="$2"
+    shift 1
+    ;;
+  '--install-yq') INSTALL_YQ=1 ;;
```

Comment:
Let's be consistent with the other `cases` and move the `INSTALL_YQ=1 ;;` statements below.